[CIS-1002] Message List Documentation (#1437)
GetStream/stream-chat-swift: docusaurus/docs/iOS/components/message-composer.md

import ComposerViewProperties from '../common-content/reference-docs/stream-chat-ui/composer/composer-view-properties.md'
import ComposerContentProperties from '../common-content/reference-docs/stream-chat-ui/composer/composer-vc.content-properties.md'

> The Message Composer provides all the UI and necessary functionality for writing and sending messages. It supports sending text, handling chat commands, autocomplete suggestions, uploading attachments like images, files, and video. The composer is a combination of two components, the `ComposerVC` and the `ComposerView`, the first one is a view controller responsible for the functionality of the composer, where the latter is only responsible for the UI layout.

Reviewer: ```suggestion
The Message Composer provides all the UI and necessary functionality for writing and sending messages. It supports sending text, handling chat commands, suggestions autocompletion, uploading attachments like images, files, and videos. The composer is a combination of two components, the `ComposerVC` and the `ComposerView`, the first one is a view controller responsible for the functionality, where the latter is only responsible for the UI and layout.
```

## Composer View Controller

The `ComposerVC` is the view controller that manages all the functionality and interaction with the `ComposerView`.

### Usage

> The `ComposerVC` is used by both the Channel and Thread components, but you can also add the `ComposerVC` in your View Controller as a child view if needed. Please keep in mind that if you do so you will need to manage the keyboard yourself. Here is an example of how you can add the composer as a child view controller:

Reviewer: ```suggestion
The `ComposerVC` is used by both the Channel and Thread components, but you can also add the `ComposerVC` as a child to your UIViewController subclass. Here is an example:
```

    override open func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)

        keyboardHandler.start()
    }

    override open func viewDidDisappear(_ animated: Bool) {
        super.viewDidDisappear(animated)

        keyboardHandler.stop()
    }
}
```

> As you can see if you want to use the `ComposerVC` in your custom message list view you need to setup the dependencies of the composer, add it as a child view controller of your custom message list view controller, and even manage the keyboard yourself or use our keyboard observer to manage it.

Reviewer: Do we need this warning?

Author: It think it is fine 🤔

### Customization

> The `ComposerVC` and `ComposerView` are completely customizable. You can not only change the UI layout and styling, but you can extend the composer functionality as well. In case you want to change the styling, adding new views, and new functionality you can take a look at the [Customize Message Composer](../guides/customize-message-composer) guide. If you want to add a new custom attachment and make the composer support it, you should read the [Message Composer Custom Attachments](../guides/working-with-custom-attachments) guide.

Reviewer: ```suggestion
The `ComposerVC` and `ComposerView` are highly customizable in both styling and functionality directions. In case you want to change the styling, adding new views, and new functionality you can take a look at the [Customize Message Composer](../guides/customize-message-composer) guide. If you think about introducing new custom attachment and make the composer support it, read the [Message Composer Custom Attachments](../guides/working-with-custom-attachments) guide.
```

### Properties

The complete list of all the `ComposerVC`'s components.

<ComposerProperties/>

## Composer View

> The `ComposerView` class holds all the composer subviews and implements the composer layout. The composer layout is built with multiple `ContainerStackView`'s, which are very similar how `UIStackView`'s work, you can read more about them [here](../customization/custom-components#setuplayout). This makes it very customizable since to change the layout you only need to move/remove/add views from different containers.

Reviewer: ```suggestion
The `ComposerView` class holds all the composer subviews and implements the composer layout. The composer layout is built with multiple `ContainerStackView`'s, which work similar to `UIStackView`'s. This makes it very customizable since to change the layout you only need to move/remove/add views to different containers. Read more about `ContainerStackView` [here](../customization/custom-components#setuplayout).
```

## Composer Content

The `ComposerVC.Content` is a struct that contains all the data that will be part of the composed message. It contains the current `text` of the message, the `attachments`, the `threadMessage` in case you are inside a Thread, the `command` if you are sending, for example, a Giphy, and the `state` of the composer to determine whether you are creating, editing or quoting a message.

> Some of the composer's content properties are mutable, like the `attachments`, `threadMessage`, `text`, and `command` properties. They can be directly changed since they represent data that do not depend on the state of the composer. On the other hand, some properties are immutable, and can only be changed through mutating functions on the `ComposerVC.Content`. This is to protect against bad states, for example, having the `editingMessage` property to `nil` but the `state = .edit`.

Reviewer: ```suggestion
Some of the composer's content properties are mutable, like the `attachments`, `threadMessage`, `text`, and `command`. They can be directly changed since they represent data that do not depend on the state of the composer. On the other hand, some properties are immutable, and can only be changed through mutating functions on the `ComposerVC.Content`. This protects a composer against bad states, for example, having the `editingMessage` property to `nil` but the `state = .edit`.
```

### State

> The composer has three different states, `.new`, `.edit` and `.quote`. The `.new` state is when the composer is creating a new message, the `.edit` state is when we are editing an existing message and changing its content, and finally, the `.quote` state is when we are replying to a message inline (not in a thread). In the table below we can see the composer in all three different states:

Reviewer: ```suggestion
The composer has three states: `.new`, `.edit`, and `.quote`. The `.new` state is when the composer is creating a new message, the `.edit` state is when we are editing an existing message and changing its content, and finally, the `.quote` state is when we are replying to a message inline (not in a thread). In the table below we can see the composer in all three different states:
```
